parts/k8s/containeraddons/kubernetesmasteraddons-metrics-server-deployment.yaml

```diff
@@ -126,7 +126,6 @@ spec:
         imagePullPolicy: IfNotPresent
         command:
         - /metrics-server
-        - --source=kubernetes.summary_api:''
```

Comment:
@jackfrancis how do we handle command parameter change in kubernetesmasteraddons_xxx.yaml file in aks-engine?

I am planning to upgrade `metrics-server` from v0.2.0 to v0.3.0 in k8s v1.15.0, while that require command parameters change, do you know what's the correct way to do this? One way could be move original `kubernetesmasteraddons-metrics-server-deployment.yaml` file to new folder like `1.14`, `1.13`, `1.12`, ... , looks like I need to create a lot of such version folder, is that the correct way?

Reply:
Correct, the current way to define k8s version-specific changes is to create a new directory with the minor.major version of the first version that these changes work for. If there is only a 1.15 directory, for example (and not a 1.16 directory), then the logic assumes the spec is valid for 1.15 and above. So every time there is a forward looking version-breaking change, we create a new dir to store those with the same name as the version that introduces the change.

Hope that makes sense!